src/test/java/uk/gov/hmcts/reform/wastandalonetaskbpmn/bpmn/CamundaCreateTaskTest.java

```diff
@@ -18,4 +28,77 @@ public void should_create_a_task_with_no_delay_until() {
         createTask(false);
     }
 
+    @Test
+    public void should_not_create_a_task_with_different_TenantId_multiple_resource() {
+
+        clearDeployments();
+
+        RepositoryService repositoryService = processEngineRule.getRepositoryService();
+
+        DeploymentBuilder deploymentBuilder = repositoryService.createDeployment()
+            .tenantId("someTenantId");
+        deploymentBuilder.addClasspathResource("wa-task-initiation-ia-asylum.bpmn")
+            .deploy();
+
+        DeploymentBuilder deploymentBuilder2 = repositoryService.createDeployment()
+            .tenantId("someTenantId2");
+
+        deploymentBuilder2.addClasspathResource("wa-task-initiation-ia-asylum_duplicate.bpmn")
+            .deploy();
+
+        assertThrows(MismatchingMessageCorrelationException.class, () -> {
+            createTask(false);
+        });
+    }
+
+    @Test
+    public void should_create_a_task_with_and_without_TenantId_multiple_resources() {
+
+        clearDeployments();
+
+        RepositoryService repositoryService = processEngineRule.getRepositoryService();
+
+        DeploymentBuilder deploymentBuilder = repositoryService.createDeployment()
+            .tenantId("someTenantId");
+        deploymentBuilder.addClasspathResource("wa-task-initiation-ia-asylum.bpmn")
+            .deploy();
+
+        DeploymentBuilder deploymentBuilder2 = repositoryService.createDeployment();
+
+        deploymentBuilder2.addClasspathResource("wa-task-initiation-ia-asylum_duplicate.bpmn")
+            .deploy();
+
+        ProcessInstance processInstance =  createTask(false);
+
+        BpmnAwareTests.complete(BpmnAwareTests.task("processTask"));
+        assertEquals(null,processInstance.getTenantId());
+        BpmnAwareTests.assertThat(processInstance).isEnded();
+    }
+
+    @Test
+    public void should_not_create_a_task_withoutTenantId_multiple_resource() {
```

Comment:
mixing snakeCase and camelCase. As it is a test name this can be tenant_id.

Reply:
Made changes as suggested.